tests/lib/TestUtils.hpp

```diff
@@ -0,0 +1,93 @@
+#ifndef TEST_UTILS_HPP
+#define TEST_UTILS_HPP
+
+#include <string.h>
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#ifdef _WIN32
+#include <windows.h>
+#else
+#include <sys/types.h>
+#include <unistd.h>
+#endif
+
+#include <webots/Emitter.hpp>
+#include <webots/Robot.hpp>
+
+using namespace webots;
+
+const std::string TEST_OUTPUT_FILENAME = "../../../output.txt";
+
+class TestUtils {
+public:
+  TestUtils(Robot *robot, const std::string &testName) : mRobot(robot), mEmitter(NULL) {
+    assert(robot);
+    // remove the full path
+    std::size_t index = testName.find_last_of("/\\");
+    mTestName = testName.substr(index + 1);
+    // keep the basename (remove the .exe extension)
+    index = mTestName.find_last_of(".");
+    mTestName = mTestName.substr(0, index);
+    notifyControllerStatus(true);
+  }
+
+  void notifyControllerStatus(bool running) {
+    if (!mEmitter)
+      mEmitter = mRobot->getEmitter("ts_emitter");
+
+    char msg[256];
+#ifdef _WIN32
+    sprintf(msg, "ts %d %ld", running, GetCurrentProcessId());
+#else
+    sprintf(msg, "ts %d %d", running, getpid());
+#endif
+    mEmitter->send(msg, strlen(msg));
+  }
+
+  void sendSuccess() {
+    // write the result to the file
+    std::ofstream file;
+    file.open(TEST_OUTPUT_FILENAME, std::ofstream::out | std::ofstream::app);
+    file << "OK: " << mTestName << std::endl;
+    file.close();
+
+    // write the result to the console
+    std::cout << "OK: " << mTestName << std::endl;
+    notifyControllerStatus(false);
+
+    delete mRobot;
+    exit(EXIT_SUCCESS);
+  }
+
+  // please use rather the assertion tests that this function
+  void sendErrorAndExit(const std::string &errorMessage) {
```

Comment:
So why not making this method private instead of writing such a comment?

Reply:
Because I copied it as-is from the C library.
I will make it private.